### PR TITLE
Change the API to support filesystem::path if this feature is available

### PR DESCRIPTION
**Pull request by dimitri-tdg**
When using unicode path under windows, there are some issues and files couldn't not be retrieved/accessed correctly. The way to fix this is to use filesystem::path to delegate path encoding to std library.

### CODE REVIEW

File: include/fx/gltf.h [hunk 1]
```diff
@@ -27,6 +27,28 @@
...
+        #define  FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #endif
+#endif
+
+#ifdef FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #include <experimental/filesystem>
+    #define FX_GLTF_FILESYSTEM std::experimental::filesystem::v1
+#else
+    #include <filesystem>
+    #define FX_GLTF_FILESYSTEM std::filesystem
```
Comment (jessey-git) [inline include/fx/gltf.h:49]: Can you undefine both of these defines at the very bottom of this file.  I try not to let any defines escape outside the header file so folks don't start depending on them.

I know that means they might have to do a similar dance in order to load the right filesystem header but I think that's actually a good thing for them to do on their side.

**Comment by jessey-git** [comment]
@dimitri-tdg Wow, thanks for doing this work! I think this looks fine as-is with 1 minor comment that I left relating to undefining the defines. I'll need to update the viewer application to account for the breaking change and I'll want to change my .clang-format to use indention for the #defines but I'll check those in afterwards on my own I think.

File: include/fx/gltf.h (hunk 1 above)
Comment (dimitri-tdg) [reply include/fx/gltf.h:49]: Thanks for your suggestion @jessey-git! I didn't see the undef at the end of the file. I've modified my commit to harmonize the code.

**Comment by dimitri-tdg** [comment]
Oh you're right @jessey-git! I didn't modify the viewer application to support this modification. I focused on unit tests and validation tests only. I can check to integrate this into the same PR but I'm not super comfortable with Windows graphical API :sweat_smile: 

Anyway, if you want to pack all modifications (code + utests + viewer application), feel free to modify/reuse my proposal!

**Pull request merged**